hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,822 @@
+                in self.xray_lines if xray_line.active]
+
+    def add_family_lines(self, xray_lines='from_elements'):
+        """Create the Xray-lines instances and configure them appropiately
+
+        If a X-ray line is given, all the the lines of the familiy is added.
+        For instance if Zn Ka is given, Zn Kb is added too. The main lines
+        (alpha) is added to self.xray_lines
+
+        Parameters
+        -----------
+        xray_lines: {None, 'from_elements', list of string}
+            If None, if `metadata` contains `xray_lines` list of lines use
+            those. If 'from_elements', add all lines from the elements contains
+            in `metadata`. Alternatively, provide an iterable containing
+            a list of valid X-ray lines symbols. (eg. ('Al_Ka','Zn_Ka')).
+        """
+
+        only_one = False
+        only_lines = ("Ka", "La", "Ma")
+
+        if xray_lines is None or xray_lines == 'from_elements':
+            if 'Sample.xray_lines' in self.spectrum.metadata \
+                    and xray_lines != 'from_elements':
+                xray_lines = self.spectrum.metadata.Sample.xray_lines
+            elif 'Sample.elements' in self.spectrum.metadata:
+                xray_lines = self.spectrum._get_lines_from_elements(
+                    self.spectrum.metadata.Sample.elements,
+                    only_one=only_one,
+                    only_lines=only_lines)
+            else:
+                raise ValueError(
+                    "No elements defined, set them with `add_elements`")
+
+        components_names = [xr.name for xr in self.xray_lines]
+        xray_lines = filter(lambda x: x not in components_names, xray_lines)
+        xray_lines, xray_not_here = self.spectrum.\
+            _get_xray_lines_in_spectral_range(xray_lines)
+        for xray in xray_not_here:
+            print("Warning: %s is not in the data energy range." % (xray))
+
+        for i, xray_line in enumerate(xray_lines):
+            element, line = utils_eds._get_element_and_line(xray_line)
+            line_energy, line_FWHM = self.spectrum._get_line_energy(
+                xray_line,
+                FWHM_MnKa='auto')
+            component = create_component.Gaussian()
+            component.centre.value = line_energy
+            component.sigma.value = line_FWHM / 2.355
+            component.centre.free = False
+            component.sigma.free = False
+            component.name = xray_line
+            self.append(component)
+            self.xray_lines.append(component)
+            init = True
+            if init:
+                self[xray_line].A.map[
+                    'values'] = self.spectrum[..., line_energy].data * \
+                    line_FWHM / self.spectrum.axes_manager[-1].scale
+                self[xray_line].A.map['is_set'] = (
+                    np.ones(self.spectrum[..., line_energy].data.shape) == 1)
+
+            component.A.ext_force_positive = True
```

Comment:
I would guess that you're correct because you're better at this stuff than me - but I can't see how to do it immediately... I'll come back to it if we want this done before merging.